plugin/field_data_utils.go

```diff
@@ -0,0 +1,105 @@
+package gcpsecrets
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault-plugin-secrets-gcp/plugin/util"
+	"github.com/hashicorp/vault/sdk/framework"
+)
+
+type inputParams struct {
+	name       string
+	secretType string
+
+	hasBindings bool
+	rawBindings string
+	bindings    ResourceBindings
+
+	project             string
+	serviceAccountEmail string
+
+	scopes []string
+}
+
+func (input *inputParams) parseOkInputSecretType(d *framework.FieldData) (warnings []string, err error) {
```

Comment:
Is there a particular reason we want the return values to be named? I also don't see `warnings` set in many of these parsing functions (aside from `parseOkInputTokenScopes`).

Reply:
> Is there a particular reason we want the return values to be named? 

I don't quite understand what you mean.

> I also don't see warnings set in many of these parsing functions (aside from parseOkInputTokenScopes).

Emily wrote the original code. I would keep it this way as well, so that there's consistency in how the `parseOk` functions are called.

Reply:
That's fair. I was referring to `(warnings []string, err error)` being changed to `([]string, error)` or even `error` in the helper functions where no `warnings` are set or returned. These are internal helpers though so if it makes them easier to read and consistent I'm fine with leaving as-is.

Reply:
I think I'll leave it in as-is for now.